tools/Dashboard/DevHome.Dashboard/Services/WidgetIconService.cs

```diff
@@ -21,103 +20,61 @@ public class WidgetIconService : IWidgetIconService
 
     private readonly WindowEx _windowEx;
 
-    private readonly IWidgetHostingService _widgetHostingService;
+    private readonly ConcurrentDictionary<string, BitmapImage> _widgetLightIconCache;
+    private readonly ConcurrentDictionary<string, BitmapImage> _widgetDarkIconCache;
 
-    private readonly Dictionary<string, BitmapImage> _widgetLightIconCache;
-    private readonly Dictionary<string, BitmapImage> _widgetDarkIconCache;
-
-    public WidgetIconService(WindowEx windowEx, IWidgetHostingService widgetHostingService)
+    public WidgetIconService(WindowEx windowEx)
     {
         _windowEx = windowEx;
-        _widgetHostingService = widgetHostingService;
 
-        _widgetLightIconCache = new Dictionary<string, BitmapImage>();
-        _widgetDarkIconCache = new Dictionary<string, BitmapImage>();
+        _widgetLightIconCache = new ConcurrentDictionary<string, BitmapImage>();
+        _widgetDarkIconCache = new ConcurrentDictionary<string, BitmapImage>();
     }
 
-    /// <summary>
-    /// Caches icons for all widgets in the WidgetCatalog that are included in Dev Home.
-    /// </summary>
-    public async Task CacheAllWidgetIconsAsync()
+    public void RemoveIconsFromCache(string definitionId)
     {
-        var cacheTasks = new List<Task>();
-        var widgetCatalog = await _widgetHostingService.GetWidgetCatalogAsync();
-        var widgetDefinitions = await Task.Run(() => widgetCatalog?.GetWidgetDefinitions());
-        foreach (var widgetDef in widgetDefinitions ?? Array.Empty<WidgetDefinition>())
-        {
-            var task = AddIconsToCacheAsync(widgetDef);
-            cacheTasks.Add(task);
-        }
-
-        await Task.WhenAll(cacheTasks);
+        _widgetLightIconCache.TryRemove(definitionId, out _);

```

Comment:
No logging if the widget can't be removed?

Reply:
I don't care if it doesn't work -- if the user never switched themes, one of the caches would be empty.